spec/examples/by_set_pos_spec.rb

```diff
@@ -0,0 +1,42 @@
+require File.dirname(__FILE__) + '/../spec_helper'
+
+module IceCube
+
+  describe MonthlyRule, 'BYSETPOS' do
+    it 'should behave correctly' do
+      schedule = IceCube::Schedule.from_ical "RRULE:FREQ=MONTHLY;COUNT=4;BYDAY=WE;BYSETPOS=4"
+      schedule.start_time = Time.new(2015, 5, 28, 12, 0, 0)
+      expect(schedule.occurrences_between(Time.new(2015, 01, 01), Time.new(2017, 01, 01))).to eq([
+                                                                                                     Time.new(2015,6,24,12,0,0),
+                                                                                                     Time.new(2015,7,22,12,0,0),
+                                                                                                     Time.new(2015,8,26,12,0,0),
+                                                                                                     Time.new(2015,9,23,12,0,0)
+                                                                                                 ])
+    end
+
+    context 'when billing occurs at the end of the month' do
+      it 'should return the correct end of the months dates' do
+        schedule = IceCube::Schedule.from_ical "RRULE:FREQ=MONTHLY;COUNT=4;BYMONTHDAY=28,29,30,31;BYSETPOS=-1"
+        schedule.start_time = Time.new(2022, 11, 1, 12, 0, 0)
+        expect(schedule.occurrences_between(Time.new(2022, 10, 01), Time.new(2023, 10, 31))).to eq([
+                                                                                                     Time.new(2022,11,30,12,0,0),
+                                                                                                     Time.new(2022,12,31,12,0,0),
+                                                                                                     Time.new(2023,01,31,12,0,0),
+                                                                                                     Time.new(2023,02,28,12,0,0)
+                                                                                                   ])
+      end
+    end
+
+  end
+
+  describe YearlyRule, 'BYSETPOS' do
+    it 'should behave correctly' do
+      schedule = IceCube::Schedule.from_ical "RRULE:FREQ=YEARLY;BYMONTH=7;BYDAY=SU,MO,TU,WE,TH,FR,SA;BYSETPOS=-1"
+      schedule.start_time = Time.new(1966,7,5)
+      expect(schedule.occurrences_between(Time.new(2015, 01, 01), Time.new(2017, 01, 01))).to eq([
+                                                                                                     Time.new(2015, 7, 31),
+                                                                                                     Time.new(2016, 7, 31)
+                                                                                                 ])
+    end
```

Comment:
Suggestion for clarity:
```suggestion
  describe YearlyRule, 'BYSETPOS' do
    subject(:schedule) { IceCube::Schedule.from_ical(from_ical) }

    let(:from_ical) { "RRULE:FREQ=YEARLY;BYMONTH=7;BYDAY=SU,MO,TU,WE,TH,FR,SA;BYSETPOS=-1" }

    before(:each) do
      schedule.start_time = schedule_start_time
    end

    let(:occurrences)         { schedule.occurrences_between(from_time, to_time) }
    let(:schedule_start_time) { Time.new(1966, 7, 5) }
    let(:from_time)           { Time.new(2015, 1, 1) }
    let(:to_time)             { Time.new(2017, 1, 1) }

    it "returns only the last day of each July in 2015 and 2016" do
      expect(occurrences).to eq(
        [
          Time.new(2015, 7, 31),
          Time.new(2016, 7, 31)
        ]
      )
    end
```

Reply:
However, I think it would be better to not use _every_ day of the week, because it doesn't exactly show that `BYSETPOS` is doing its job.

Perhaps,
```ruby
    let(:from_ical) { "RRULE:FREQ=YEARLY;BYMONTH=7;BYDAY=SU,SA;BYSETPOS=-1" }
```
otherwise, I think leave out `BYDAY`, but then it's not really testing `BYSETPOS` exactly, AFAIU.  Maybe this needs more examples:

- `BYDAY=SU,MO,TU,WE,TH,FR,SA`
- `BYDAY=SU,SA`
- `BYDAY=MO,TU,WE,TH,FR`
- `BYDAY=FR,TH,WE,TU,MO`